_posts/2017-12-18-Meeseeks-Box-DIY-Guide.md

```diff
@@ -0,0 +1,273 @@
+If the LED needs 3 volts to function and uses 20 milliamps doing so:  
+R = U / I = 5V (pin out) - 3V (LED) /  0,02A = 2V / 0,02A = 100Ohm  
+This means that a 100Ohm resistor needs to be put in series with the LED to prevent it from causing any damage to the IO pin/circuitry.
+
+*main.js*
+```javascript
+function doCall(method) {
+    var request = http.request({
+        host: 'rnm-meeseeks-box.cfapps.io',
+        port: 443,
+        path: '/',
+        method: method,
+        headers: {
+            'Content-Type': 'application/json',
+            'Content-Length': 0
+        }
+    });
+
+    request.write('');
+    request.end();
+}
+```
+The code above is a simple snippet used to make a call with no contents to the remote server.
+When the URL is called via the POST method a meeseeks is created.
+When the URL is called via the DELETE method the currently active meeseekses are destroyed (for testing).
+You can edit this to perform any action you like.
+
+<br/>
+A video showing a fully operational meeseeks box:
+<video class="image-fit" width="100%" controls>
+  <source src="/img/meeseeks/video.mp4" type="video/mp4">
+</video>
+<br/><br/>
+
+
+### Running Node as a systemd service on linux
+As the meeseeks box needs to be simple to use, the application should automatically start when the Pi does.
+The best option was to make a systemd service and make that start on startup.
+
+First we need to create the systemd service file:  
+```
+sudo nano /lib/systemd/system/meeseeks.service
+```
+This will create a new file (if one does not exist yet).
+Place the contents below in this file and save it.
+
+```
+[Unit]
+Description=Meeseeks box service
+After=network.target
+
+[Service]
+Type=simple
+User=your-user-here
+ExecStart=/usr/bin/node /home/meeseeks/main.js
+Restart=on-failure
+
+[Install]
+WantedBy=multi-user.target
+```
+This file tells systemd what the service is and does, with what executable and which user.
+The `After=network.target` tells the service daemon that this service should only start is the network stack has already loaded!
+
+To test the service first execute: `sudo systemctl daemon-reload`
+This reloads the daemon so it knows of the newly created service.
+Now we can manually start/stop/reload the service by using: `sudo systemctl start meeseeks` where you swap out `start` with the action you want to perform.
+
+To make the service run at startup use: `sudo systemtl enable meeseeks` and to disable it again, use the same command but swap out `enable` for `disable`.
+
+A far more detailed explenation about this matter can be found <a href="http://nodesource.com/blog/running-your-node-js-app-with-systemd-part-1/">here.</a>
+
+
+## Meeseeks at Devoxx
+The entire purpose of the meeseeks box was to be part of our booth at the well known Devoxx conference in Belgium.
+Our booth drew quite the crowd this year, mostly because of the nachos and the totally real szechaun sauce to go with them. 
```

Comment:
Szechuan

Hmm, it was just nacho sauce or was that meant sarcastically? :-)

Reply:
sarcastically ;)